sql/create.go

```diff
@@ -42,15 +42,29 @@ func (p *planner) CreateDatabase(n *parser.CreateDatabase) (planNode, error) {
 		}
 		return nil, fmt.Errorf("database \"%s\" already exists", n.Name)
 	}
+
 	ir, err := p.db.Inc(keys.DescIDGenerator, 1)
 	if err != nil {
 		return nil, err
 	}
-	nsID := uint32(ir.ValueInt() - 1)
+
+	desc := makeDatabaseDesc(n)
+	desc.ID = uint32(ir.ValueInt() - 1)
+
+	// TODO(pmattis): The error currently returned below is likely going to be
+	// difficult to interpret.
 	// TODO(pmattis): Need to handle if-not-exists here as well.
-	if err := p.db.CPut(nameKey, nsID, nil); err != nil {
+	err = p.db.Txn(func(txn *client.Txn) error {
+		descKey := keys.MakeDescMetadataKey(desc.ID)
+		b := &client.Batch{}
+		b.CPut(nameKey, descKey, nil)
+		b.Put(descKey, &desc)
```

Comment:
I don't recall any reason not to use `CPut` here other than it not being necessary. Note that `descKey` is going to be globally unique at this point because we just allocated the descriptor ID.